src/server/report/test.js

```diff
@@ -121,15 +236,54 @@ class Test extends LitElement {
 		const testData = fileData.tests.find(t => t.name === this.test);
 		if (!testData) return {};
 
-		return { fileData, testData };
+		const filteredBrowsers = this.browsers.split(',');
+		const browsers = data.browsers.filter(b => filteredBrowsers.includes(b.name));
+
+		return { browsers, fileData, testData };
 
 	}
+	_getSelectedBrowser(browsers, testData) {
+		let selectedBrowserIndex = this._selectedBrowserIndex;
+		if (selectedBrowserIndex > browsers.length - 1) {
+			selectedBrowserIndex = -1;
+		}
+		if (selectedBrowserIndex < 0) {
+			selectedBrowserIndex = browsers.findIndex(b => testData.results.find(r => r.name === b.name && !r.passed));
+		}
+		if (selectedBrowserIndex < 0) {
+			selectedBrowserIndex = 0;
+		}
+		const selectedBrowser = browsers[selectedBrowserIndex];
+		return selectedBrowser;
+	}
```

Comment:
```suggestion
	_getSelectedBrowser(browsers, { results }) {
		return browsers[this._selectedBrowserIndex] ||
			browsers.find(b => results.find(r => r.name === b.name && !r.passed)) ||
			browsers[0];
	}
```

Reply:
Forgot that you can access out of bounds array elements and it doesn't throw. This is... much less verbose!